### PR TITLE
Code transform to perform License and Copyright header removal

### PR DESCRIPTION
**Pull request by ykalathiya**
Here, We have used scancode.api to detect the license and copyright.

## Why are these changes needed?
This help to remove license and copyright from any language of code.

### CODE REVIEW

**Comment by daw3rd** [comment]
Please run `make conventions` in your transforms/[ray,python] directories

**Comment by daw3rd** [comment]
And please split the ray tree into ray and python runtimes.  See filter and noop projects for examples.

**Review by daw3rd — CHANGES_REQUESTED** [review]
Per earlier comments, split the ray dir into ray and python-only runtimes.
Also, after doing that, run `make conventions` in each of the python and ray directories.

**Comment by daw3rd** [comment]
https://github.com/IBM/data-prep-kit/tree/dev/transforms#transform-project-conventions will be helpful on splitting ray/python.  also, generally you may want to copy the Makefile and Dockerfile from the universal/filter/[ray,python] directories and use as templates.  Also note the various readmes.

**Review by daw3rd — CHANGES_REQUESTED** [review]
Per the build failure, you need to create a test directory with some tests.  However,  you might wait to do that until you split the ray dir into ray and python.  See one of the other transforms like universal/filter that shows how this can be done.

**Comment by ykalathiya** [comment]
@daw3rd @Param-S @Bytes-Explorer I updated the PR. Please review it.

**Comment by shivdeep-singh-ibm** [comment]
@ykalathiya You should [update your name and email  in .gitconfig](https://linuxize.com/post/how-to-configure-git-username-and-email/#setting-global-git-username-and-email)  and add sign off to your commits `git commit -s -m <some message>`. Then you will not get `DCO` Failure.

File: transforms/code/license_copyright_removal/python/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+COPY --chown=dpk:root data-processing-lib-python/ data-processing-lib-python/
+RUN cd data-processing-lib-python && pip install --no-cache-dir -e .
+
+# END OF STEPS destined for a data-prep-kit base image 
+
+COPY --chown=dpk:root src/ src/
+COPY --chown=dpk:root pyproject.toml pyproject.toml 
+RUN pip install --no-cache-dir -e .
+
+#COPY requirements.txt requirements.txt
```
Comment (Param-S) [inline transforms/code/license_copyright_removal/python/Dockerfile]: @ykalathiya i think, this needs to be uncommented. I see a docker build failure due to scancode module missing...
https://github.com/IBM/data-prep-kit/actions/runs/9560117498/job/26352464226?pr=239

**Review by ykalathiya — COMMENTED** [review]
Changed Docker file.

**Comment by ykalathiya** [comment]
I'm closing this PR due to commit sign off failures. I have created new PR. #315

**Pull request closed without merging**